src/write.c

```diff
@@ -233,6 +233,7 @@ typedef struct avifEncoderData
     uint16_t primaryItemID;
     avifEncoderItemIdArray alternativeItemIDs; // list of item ids for an 'altr' box (group of alternatives to each other)
     avifBool singleImage; // if true, the AVIF_ADD_IMAGE_FLAG_SINGLE flag was set on the first call to avifEncoderAddImage()
+    avifBool allIntra;    // if true, there was no call to avifEncoderAddImage() past the first one with AVIF_ADD_IMAGE_FLAG_NONE
```

Comment:
yeah, i agree. i think looking at `sample->sync` is the most robust way since that has the ultimate say on whether the frame is actually a keyframe or not.

also, `allIntra` is a sort of confusing term since AV1 has an "IntraOnly" frame type (a frame that only uses intra references, but it is not a keyframe since it does not reset the decoder state). so if the intention is to check for only key frames, maybe we should rename this field to `allKeyFrames`?

Reply:
According to https://aomediacodec.github.io/av1-avif/v1.1.0.html#image-and-image-collection-brand:

> Additionally, the brand avio is defined. If the file indicates the brand avio in the compatible_brands field of the FileTypeBox, then the primary item or all the items referenced by the primary item shall be AV1 image items made only of Intra Frames. Conversely, if the previous constraint applies, the brand avio should be used in the compatible_brands field of the FileTypeBox.

So I am looking for Intra Frames as defined by [AV1](https://aomediacodec.github.io/av1-spec/#terms-and-definitions) here. Key Frame is an even stronger concept but it does not matches the `avio` brand exactly. Do you know how to check for that?

My understanding is that we should use the `avio` brand in all cases (not only sequences), except for layers:
- Non-layered still images are always Intra Frames, even with grids;
- Sequences cannot be combined with layers or grids, and the first frame of the sequence (referred to by the primary image item) is always an Intra Frame.

Would that be accurate?

Reply:
>  Do you know how to check for that?

`sample->sync` is set only if the output frame is a keyframe (not for allintra frames). AFAIK, there is no easy way to check for all intra other than to inspect the frame header OBU (see `frame_type` here: https://aomediacodec.github.io/av1-spec/#uncompressed-header-semantics) of the encoded frame.

> Would that be accurate?

Yes, reading the spec, it seems like setting `avio` for all non-layered images seems like the right thing to do.

However, i don't really understand the point of this particular sub-brand/distinction. It seems largely theoretical. If you have an AV1 intra-only decoder, that decoder can figure out whether or not it can decode AVIF files by just looking at this compatible brand. But then it seems optional, and such a decoder would fail on a non-intra frame anyway.

Reply:
> Yes, reading the spec, it seems like setting avio for all non-layered images seems like the right thing to do.

Done.

> However, i don't really understand the point of this particular sub-brand/distinction. It seems largely theoretical.

Yes, this is how the specification was designed, to be generic and king of abstract I guess.

> But then it seems optional, and such a decoder would fail on a non-intra frame anyway.

This could be said for each of the 7 possible current brands.

To me the main downside is the extra 4 bytes added to most of the files generated with libavif. But if we would like to save on file size, I guess `infe`>`item_name` is informational and would be the place to start:

https://github.com/AOMediaCodec/libavif/blob/0d09ac5d60e26189820a1c3e54fb641d3432808e/src/write.c#L1047-L1051

Reply:
Done.

Reply:
Two more thoughts about the avio brand for AVIF still images.

1. To check that recommendation ("should"), ComplianceWarden will need to decode the AV1 bitstreams in the primary item and all the items referenced by the primary item. I think ComplianceWarden is unlikely to go to the extent of decoding AV1 bitstreams, so it is unlikely to warn about that recommendation.
2. Given the large number of existing AV1 still images that don't comply with that recommendation (I believe libheif doesn't add the avio brand either), I am wondering if we should update the AVIF specification to reflect the reality. If so, whether we should reword that recommendation or just remove it.